test/jdk/java/net/httpclient/ExpectContinueTest.java

```diff
@@ -0,0 +1,296 @@
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.StringTokenizer;
+import java.util.concurrent.CompletableFuture;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class ExpectContinueTest implements HttpServerAdapters {
+
+    HttpTestServer http1TestServer; // HTTP/1.1
+    Http1HangServer http1HangServer;
+    HttpTestServer http2TestServer; // HTTP/2
+
+    URI getUri;
+    URI postUri;
+    URI hangUri;
+    URI h2getUri;
+    URI h2postUri;
+    URI h2hangUri;
+
+    static final String EXPECTATION_FAILED_417 = "417 Expectation Failed";
+
+    @BeforeTest
+    public void setup() throws Exception {
+        InetSocketAddress sa = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+        InetSocketAddress saHang = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+
+        http1TestServer = HttpTestServer.of(HttpServer.create(sa, 0));
+        http1TestServer.addHandler(new GetHandler(), "/http1/get");
+        http1TestServer.addHandler(new PostHandler(), "/http1/post");
+        getUri = URI.create("http://" + http1TestServer.serverAuthority() + "/http1/get");
+        postUri = URI.create("http://" + http1TestServer.serverAuthority() + "/http1/post");
+
+        // Due to limitations of the above Http1 Server, a manual approach is taken to test the hanging with the
+        // httpclient using Http1 so that the correct response header can be returned for the test case
+        http1HangServer = new Http1HangServer(saHang);
+        hangUri = URI.create("http://" + http1HangServer.ia.getCanonicalHostName() + ":" + http1HangServer.port + "/http1/hang");
+
+
+        http2TestServer = HttpTestServer.of(
+                new Http2TestServer("localhost", false, 0));
+        http2TestServer.addHandler(new GetHandler(), "/http2/get");
+        http2TestServer.addHandler(new PostHandler(), "/http2/post");
+        http2TestServer.addHandler(new PostHandlerCantContinue(), "/http2/hang");
+        h2getUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/get");
+        h2postUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/post");
+        h2hangUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/hang");
+
+        http1TestServer.start();
+        http1HangServer.start();
+        http2TestServer.start();
+    }
+
+    @AfterTest
+    public void teardown() throws IOException {
+        http1TestServer.stop();
+        http1HangServer.close();
+        http2TestServer.stop();
+    }
+
+    static class GetHandler implements HttpTestHandler {
+
+        @Override
+        public void handle(HttpTestExchange exchange) throws IOException {
+            try (InputStream is = exchange.getRequestBody();
```

Comment:
Should this also be doing something similar to what the `PostHandler` does for HTTP2 versioned request?

Reply:
I don't think that is necessary here as in both HTTP/1 and HTTP/2 cases, the flow is exactly the same. The case you mentioned was due to the HTTP/1 server emitting a 100 response code in its implementation without it being explicitly specified in the handler.